### DISCUSSION TITLE
Kotlin data classes in Dart?

### DISCUSSION THREAD

**Discussion by tonnyavery** [dotlin-org/dotlin#109] [General]
I would like to share domain language definition between Kotlin backend and Flutter apps.
Can I use dotlin to write a very simple set of Kotlin data classes and use them in Dart as a shared library?

**Comment by wilkomanger** [comment]
The goal of Dotlin is not to make code shareable between Kotlin/JVM and Dart, but rather to bring the Kotlin language (and extra features) to the Dart runtime.

However, depending on the data classes you could still do this, if they're simple enough. You could then compile those specific files with Dotlin and you'd have some code sharing. But you might run in some trouble with stdlib definitions, for example `Int`, which is 64-bit in Dotlin/Dart and 32-bit in Kotlin.

Hope that helps :+1:
